onnxruntime/core/optimizer/graph_optimizer_registry.h

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/common/inlined_containers.h"
+#include "core/common/logging/logging.h"
+#include "core/optimizer/graph_transformer.h"
+#include "core/framework/execution_providers.h"
+#include "core/framework/compute_capability.h"
+
+namespace onnxruntime {
+using KeyValueConfig = std::unordered_map<std::string, std::string>;
+using SelectionFunc = std::function<std::vector<std::unique_ptr<ComputeCapability>>(const GraphViewer&, const KeyValueConfig&)>;
+
+/**
+ * A registration/lookup class for re-usable optimizers for EPs.
+ */
+class GraphOptimizerRegistry {
+ public:
+  explicit GraphOptimizerRegistry();
+  GraphOptimizerRegistry(const GraphOptimizerRegistry&) = delete;
+
+  /**
+   * Get GraphOptimizerRegistry instance as a singleton.
+   */
+  static std::shared_ptr<GraphOptimizerRegistry> Get() {
```

Comment:
Can graph_optimizer_registry_ be a unique_ptr instead, with `Get` returning `const GraphOptimizerRegistry&`?

I don't think the caller should be getting any ownership of the instance (via shared_ptr), nor do they need to know details about implementation details like how the instance is managed (unique_ptr). They only need the GraphOptimizerRegistry instance. 

If Create hasn't been called yet, throw.

Reply:
fixed.